client/state/editor/actions.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Internal dependencies
+ */
+import warn from 'lib/warn';
+import { POST_EDITOR_EDIT, POST_EDITOR_START } from 'state/action-types';
+import { getSitePost } from 'state/posts/selectors';
+import { getCurrentPostGlobalId, isLoadingPost } from './selectors';
+
+export function startEditing( siteId, postId ) {
+	const action = { type: POST_EDITOR_START };
+	if ( ! postId ) {
+		return Object.assign( action, { siteId } );
+	}
+
+	return ( dispatch, getState ) => {
+		const post = getSitePost( getState(), siteId, postId );
+		if ( post ) {
+			Object.assign( action, { globalId: post.globalId } );
+		} else {
+			Object.assign( action, { siteId, postId } );
+		}
+
+		dispatch( action );
+	};
+}
+
+export function editPost( post, globalId ) {
+	return ( dispatch, getState ) => {
+		const state = getState();
+
+		if ( ! globalId ) {
```

Comment:
>I strongly feel this is something we should pass explicitly and not be part of the action.

Which I think is still compatible with the overall goal of this pull request, that currently edited post be a tracked piece of state, but rather than making assumptions from the action creators, place the burden on the developer to pass the currently edited post global ID:

```js
editPost( {
	title: 'Hello World'
}, getCurrentPostGlobalId( state ) );
```